bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -139,6 +160,39 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
         }
     }
 
+    private Collection<Resource> getAssetFolders(Page page, ResourceResolver resolver) {
+        Map<String, Resource> allAssetFolders = new TreeMap<String, Resource>();
+        ValueMap properties = page.getProperties();
+        String[] configuredAssetFolderPaths = properties.get(damAssetProperty, String[].class);
+        if (configuredAssetFolderPaths != null) {
+            for (String configuredAssetFolderPath : configuredAssetFolderPaths) {
+                if (StringUtils.isNotBlank(configuredAssetFolderPath)) {
+                    Resource configuredAssetFolder = resolver.getResource(configuredAssetFolderPath);
+                    if (configuredAssetFolder != null && configuredAssetFolder.isResourceType("sling:OrderedFolder")) {
+                        for (Resource assetFolder : getAssetFolders(configuredAssetFolder)) {
+                            if (!allAssetFolders.containsKey(assetFolder.getPath())) {
+                                allAssetFolders.put(assetFolder.getPath(), assetFolder);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return allAssetFolders.values();
+    }
+
+    private List<Resource> getAssetFolders(Resource assetFolder) {
+        List<Resource> damFolders = new ArrayList<Resource>();
+        damFolders.add(assetFolder);
+        for (Iterator<Resource> children = assetFolder.listChildren(); children.hasNext();) {
+            Resource assetFolderChild = children.next();
+            if (assetFolderChild.isResourceType("sling:OrderedFolder")) {
```

Comment:
Yes, that's the exact scenario I'm preventing.  But you're absolutely right in that I should be able to simply look at the paths and filter out the dupes.  Seems like a good idea to prevent some unnecessary traversal.  I'll look into it.